# Graded assignment 1

Anthony van den Berg, 2011118


# Packages

We need the package `tidyverse`

In [16]:
library(tidyverse) 

# Assignment 1

First, we generate some random data:

In [29]:
df1 <- as.data.frame(matrix(runif(20*2), ncol=20) )
head(df1)

V1        V2        V3        V4        V5        V6         V7       
1 0.9372075 0.9444038 0.3143790 0.3571593 0.1978254 0.05983767 0.9344446
2 0.7272334 0.3906534 0.0268274 0.7826705 0.9796262 0.73050878 0.8934992
  V8        V9        V10       V11       V12       V13       V14      
1 0.3361101 0.8906456 0.1846935 0.1029586 0.1482965 0.5706998 0.5813720
2 0.9532875 0.6994921 0.1710837 0.7036887 0.0457021 0.7046277 0.5441034
  V15       V16       V17        V18       V19       V20      
1 0.7602751 0.3103151 0.97554703 0.3624438 0.8905420 0.3290343
2 0.4104692 0.7033053 0.02647392 0.9543806 0.8241318 0.8719316

_df1_ is a wide dataframe. The goal of the assignment is to change _df1_ into a long dataframe _df2_. And show the first rows of the dataframe using `head(df2)`

The dataframe _df2_ should look like this:

| Key | Value      |
|-----|------------|
| V1  | 0.09413864 |
| V1  | 0.69782526 |
| V2  | 0.23460454 |
| V2  | 0.51520514 |
| V3  | 0.63181844 |
| V3  | 0.22807211 |

In [30]:
df2 <- df1 %>%
  gather(Year, Value)

head(df2)

Key Value    
1 V1   0.9372075
2 V1   0.7272334
3 V2   0.9444038
4 V2   0.3906534
5 V3   0.3143790
6 V3   0.0268274

## Assignment 2

We will generate some data:

In [31]:
df3 <- as.data.frame(read.table(header=TRUE, text='
      id        V_1       V_2                
          1  42.467573  -87.810047     
          2  42.049148  -88.273029
          3  39.110539  -90.324080
                       '))

head(df3)

id V_1      V_2      
1 1  42.46757 -87.81005
2 2  42.04915 -88.27303
3 3  39.11054 -90.32408

In [32]:
df4 <- as.data.frame(read.table(header=TRUE, text='
     V_3            V_4         id                 
      one         1h            1
      two         1h            1
      three       8h            2
      four        1h            4
'))

head(df4)

V_3   V_4 id
1 one   1h  1 
2 two   1h  1 
3 three 8h  2 
4 four  1h  4


You will have to merge these files, returning all rows and all columns from both _df3_ and _df5_. Where there are not matching values, return _NA_ for the missings. 

This should be the end result:

| id | V_1      | V_2       | V_3   | V_4 |
|----|----------|-----------|-------|-----|
| 1  | 42.46757 | -87.81005 | one   | 1h  |
| 1  | 42.46757 | -87.81005 | two   | 1h  |
| 2  | 42.04915 | -88.27303 | three | 8h  |
| 3  | 39.11054 | -90.32408 | NA    | NA  |
| 4  | NA       | NA        | four  | 1h  |



In [33]:
df5 <- merge(df3, df4, all=TRUE)
df5

id V_1      V_2       V_3   V_4
1 1  42.46757 -87.81005 one   1h 
2 1  42.46757 -87.81005 two   1h 
3 2  42.04915 -88.27303 three 8h 
4 3  39.11054 -90.32408 NA    NA 
5 4        NA        NA four  1h

End of notebook